### PR TITLE
Add additive synthesis tutorial

### PR DESCRIPTION
**Pull request by mthrok**
This commit adds the tutorial for additive synthesis, using torchaudio's prototype DSP ops.

[Review here](https://output.circle-artifacts.com/output/job/3dc83322-832a-4272-9c13-df752c97b660/artifacts/0/docs/tutorials/additive_synthesis_tutorial.html)

### CODE REVIEW

File: examples/tutorials/additive_synthesis_tutorial.py [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+Additive Synthesis
+==================
+
+**Author**: `Moto Hira <moto@meta.com>`__
+
+This tutorial is the continuation of
+`Oscillator and ADSR Envelope <./oscillator_tutorial.html>`__.
+
+This tutorial shows how to perform additive synthesis and subtractive
+synthesis with TorchAudio's DSP functions.
```
Comment (carolineechen) [inline examples/tutorials/additive_synthesis_tutorial.py:12]:
```suggestion
synthesis using TorchAudio's DSP functions.
```

File: examples/tutorials/additive_synthesis_tutorial.py [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+#
+# The core of additive synthesis is oscillator. We create a timbre by
+# summing up the multiple waveforms generated by oscillator.
+#
+# In `the oscillator tutorial <./oscillator_tutorial.html>`__, we used
+# :py:func:`~torchaudio.prototype.functional.oscillator_bank` and
+# :py:func:`~torchaudio.prototype.functional.adsr_envelope` to generate
+# various waveforms.
+#
+# In this tutorial, we also use
```
Comment (carolineechen) [inline examples/tutorials/additive_synthesis_tutorial.py:67]:
```suggestion
# In this tutorial, we use
```

File: examples/tutorials/additive_synthesis_tutorial.py [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+# :py:func:`~torchaudio.prototype.functional.extend_pitch` to create
+# a timbre from base frequency.
+#
+
+######################################################################
+#
+# First, we define some constants and helper functions we use
```
Comment (carolineechen) [inline examples/tutorials/additive_synthesis_tutorial.py:74]:
```suggestion
# First, we define some constants and helper function that we use
```

File: examples/tutorials/additive_synthesis_tutorial.py [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+    if zoom is not None:
+        ax = fig.add_axes([pos.x0 + 0.02, pos.y0 + 0.03, pos.width / 2.5, pos.height / 2.0])
+        ax.plot(t, waveform)
+        ax.set(xlim=zoom, xticks=[], yticks=[])
+
+    waveform /= waveform.abs().max()
+    return Audio(vol * waveform, rate=sample_rate, normalize=False)
+
+######################################################################
+# Harmonics Overtones
```
Comment (carolineechen) [inline examples/tutorials/additive_synthesis_tutorial.py:125]:
```suggestion
# Harmonic Overtones
```

File: examples/tutorials/additive_synthesis_tutorial.py [hunk 5]
```diff
@@ -0,0 +1,326 @@
...
+# -------------------
+#
+# Harmonic overtones are frequency components that are integer
```
Comment (carolineechen) [inline examples/tutorials/additive_synthesis_tutorial.py:128]:
```suggestion
# Harmonic overtones are frequency components that are an integer
```

File: examples/tutorials/additive_synthesis_tutorial.py [hunk 6]
```diff
@@ -0,0 +1,326 @@
...
+#  - Triangle wave
+#
+
+######################################################################
+# Sawtooth wave
+# ~~~~~~~~~~~~~
+#
+# `Sawtooth wave <https://en.wikipedia.org/wiki/Sawtooth_wave>`_ can be
+# expressed as the following. It contains all the integer harmonics, so
+# it is commonly sued in subtractive synthesis as well.
```
Comment (carolineechen) [inline examples/tutorials/additive_synthesis_tutorial.py:145]:
```suggestion
# it is commonly used in subtractive synthesis as well.
```

File: examples/tutorials/additive_synthesis_tutorial.py [hunk 7]
```diff
@@ -0,0 +1,326 @@
...
+
+
+######################################################################
+#
+# Now synthesize a waveform
+#
+
+freq0 = torch.full((NUM_FRAMES, 1), F0)
+amp0 = torch.ones((NUM_FRAMES, 1))
+freq, amp, waveform = sawtooth_wave(freq0, amp0, int(SAMPLE_RATE / F0), SAMPLE_RATE)
```
Comment (carolineechen) [inline examples/tutorials/additive_synthesis_tutorial.py:178]: just an fyi, a couple times in the tutorial there is a user warning `Some frequencies are above nyquist frequency. Setting the corresponding amplitude to zero. This might cause numerically unstable gradient.`

File: examples/tutorials/additive_synthesis_tutorial.py [hunk 8]
```diff
@@ -0,0 +1,326 @@
...
+freq0 = torch.full((NUM_FRAMES, 1), F0)
+amp0 = torch.ones((NUM_FRAMES, 1))
+freq, amp, waveform = square_wave(freq0, amp0, int(SAMPLE_RATE/F0/2), SAMPLE_RATE)
+show(freq, amp, waveform, SAMPLE_RATE, zoom=(1/F0, 3/F0))
+
+######################################################################
+# Triangle wave
+# ~~~~~~~~~~~~~
+#
+# `Triangle wave <https://en.wikipedia.org/wiki/Triangle_wave>`_.
```
Comment (carolineechen) [inline examples/tutorials/additive_synthesis_tutorial.py:236]:
```suggestion
# `Triangle wave <https://en.wikipedia.org/wiki/Triangle_wave>`_
```

File: examples/tutorials/additive_synthesis_tutorial.py [hunk 9]
```diff
@@ -0,0 +1,326 @@
...
+show(freq, amp, waveform, SAMPLE_RATE, zoom=(1/F0, 3/F0))
+
+######################################################################
+# Inharmonic Paritials
+# --------------------
+#
+# Inharmonic partials refer to freqencies that are not integer multiple
+# of fundamental frequency.
+#
+# They are essential in re-creating the real sound from our lives or
```
Comment (carolineechen) [inline examples/tutorials/additive_synthesis_tutorial.py:275]:
```suggestion
# They are essential in re-creating real sounds from our lives or
```

File: examples/tutorials/additive_synthesis_tutorial.py (hunk 7 above)
Comment (mthrok) [reply examples/tutorials/additive_synthesis_tutorial.py:178]: Yeah, this happens because we are filling the entire frequency range with frequency series some of which exceed the Nyquist frequency. We could suppress the the warning, but I thought it's overkill and since we do not do any back propagation in this tutorial, so I think it's okay. We have occasional warnings (like the one from PyTorch core when creating a Tensor from read-only buffer)

File: examples/tutorials/additive_synthesis_tutorial.py (hunk 9 above)
Comment (mthrok) [reply examples/tutorials/additive_synthesis_tutorial.py:275]:
```suggestion
# They are essential in re-creating realistic sound or
```

**Review by carolineechen — APPROVED** [review]
